Sources/ApolloCore/Atomic.swift

```diff
@@ -22,10 +22,11 @@ public class Atomic<T> {
   
   /// Mutates the underlying value within a lock.
   /// - Parameter block: The block to execute to mutate the value.
-  public func mutate(block: (inout T) -> Void) {
+  public func mutate<U>(block: (inout T) -> U) -> U {
```

Comment:
worth making this a `@discardableResult` so if we don't need it, we don't have to explicitly discard it? 

Also can you please update the doc comment to reflect what the return value is? Thank you!

Reply:
I did update the documentation.

I would advise against `@discardableResult` unless you really want it. By not making the result discardable, this method behaves the same as standard methods such as [DispatchQueue.sync(_:)](https://developer.apple.com/documentation/dispatch/dispatchqueue/2016081-sync), [autoreleasepool(invoking:)](https://developer.apple.com/documentation/objectivec/2299644-autoreleasepool/), and many others.

Not using `@discardableResult` makes sure Apollo developers get a warning when they write, for example:

```swift
// Expectable compiler warning: the value is unused.
myAtomic.mutate {
    $0 = true
    return <some value>
}
```

Not using `@discardableResult` does not create any friction when `U` is `Void`: the Swift compiler is smart enough to not emit any warning:

```swift
// No warning
myAtomic.mutate { $0 = true }
```

Sometimes, as in 1f355c91ec500f99d1a4ddcdddcfe75c4a82f62c, one will have to insert a `_` in order to silence a warning. One would do the same in the context of `Dispatch.sync` and other similar methods. So not using `@discardableResult` does not add any inconvenience on top of the standard ones.

Reply:
> Not using @discardableResult does not create any friction when the `U` generic type is `Void`

I think this alleviates most of my concerns since this covers a pretty solid majority of cases.